ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -279,14 +299,18 @@ def take_step(self, env: BaseUnityEnvironment, curr_info: AllBrainInfo):
 
         # If any lessons were incremented or the environment is
         # ready to be reset
-        if self.meta_curriculum and any(lessons_incremented.values()):
-            curr_info = self._reset_env(env)
-            for brain_name, trainer in self.trainers.items():
-                trainer.end_episode()
-            for brain_name, changed in lessons_incremented.items():
-                if changed:
-                    self.trainers[brain_name].reward_buffer.clear()
+        meta_curriculum_reset = (self.meta_curriculum) and any(lessons_incremented.values())
```

Comment:
nit: don't really need to check `self.meta_curriculum` here - lessons_incremented will always be empty if it's None